core/trino-main/src/test/java/io/trino/sql/planner/TestDynamicFilter.java

```diff
@@ -293,156 +296,173 @@ public void testCrossJoinInequalityWithCastOnTheRight()
-                                                ImmutableList.of(new DynamicFilterPattern("O_COMMENT", LESS_THAN, "expr")),
-                                                filter(
-                                                        TRUE_LITERAL,
-                                                        tableScan("orders", ImmutableMap.of("O_COMMENT", "comment"))),
-                                                anyTree(
-                                                        project(
-                                                                ImmutableMap.of("expr", expression("CAST(L_COMMENT AS varchar(79))")),
-                                                                tableScan("lineitem",
-                                                                        ImmutableMap.of("L_COMMENT", "comment")))))))));
+                                        innerJoin()
+                                                .dynamicFilter(ImmutableList.of(new DynamicFilterPattern("O_COMMENT", LESS_THAN, "expr")))
+                                                .left(
+                                                        filter(
+                                                                TRUE_LITERAL,
+                                                                tableScan("orders", ImmutableMap.of("O_COMMENT", "comment"))))
+                                                .right(
+                                                        anyTree(
+                                                                project(
+                                                                        ImmutableMap.of("expr", expression("CAST(L_COMMENT AS varchar(79))")),
+                                                                        tableScan("lineitem",
+                                                                                ImmutableMap.of("L_COMMENT", "comment")))))
+                                                .build()))));
     }
 
     @Test
     public void testJoin()
     {
         assertPlan("SELECT o.orderkey FROM orders o, lineitem l WHERE l.orderkey = o.orderkey",
                 anyTree(
-                        join(
-                                INNER,
-                                ImmutableList.of(equiJoinClause("ORDERS_OK", "LINEITEM_OK")),
-                                ImmutableMap.of("ORDERS_OK", "LINEITEM_OK"),
-                                anyTree(
-                                        tableScan("orders", ImmutableMap.of("ORDERS_OK", "orderkey"))),
-                                exchange(
-                                        project(
-                                                tableScan("lineitem", ImmutableMap.of("LINEITEM_OK", "orderkey")))))));
+                        innerJoin()
+                                .equiCriteria("ORDERS_OK", "LINEITEM_OK")
+                                .dynamicFilter(ImmutableMap.of("ORDERS_OK", "LINEITEM_OK"))
```

Comment:
seems like single DF case is super common. We could optimize builder for it, so that
```
.dynamicFilter("ORDERS_OK", "LINEITEM_OK")
```
is sufficient